### PR TITLE
[CDAP-7952] metadata search returns 400 when query omitted

### PR DESCRIPTION
**Pull request by liu-joe**

### CODE REVIEW

**Comment by sreevatsanraman** [comment]
Add a test case

File: cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataHttpHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpVersion;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Tests for {@link MetadataHttpHandler}.
+ */
+public class MetadataHttpHandlerTest {
```
Comment (anwar6953) [inline cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataHttpHandlerTest.java]: Can we just leverage `MetadataHttpHandlerTestRun`?

Comment (liu-joe) [reply cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataHttpHandlerTest.java]: Didn't know this existed, I'll update it

**Review by albertshau — APPROVED** [review]
lgtm

**Comment by liu-joe** [comment]
http://builds.cask.co/browse/CDAP-RUT666-1

**Pull request merged**